internals/models.py

```diff
@@ -924,12 +930,11 @@ def new_crbug_url(self):
       params.append('cc=' + ','.join(self.owner))
     return url + '?' + '&'.join(params)
 
-  def __init__(self, *args, **kwargs):
-    super(Feature, self).__init__(*args, **kwargs)
+  def _pre_put_hook(self):
 
     # Stash existing values when entity is created so we can diff property
     # values later in put() to know what's changed. https://stackoverflow.com/a/41344898
-    for prop_name, prop in self.properties().iteritems():
+    for prop_name, prop in self._properties.iteritems():
       old_val = getattr(self, prop_name, None)
       setattr(self, '_old_' + prop_name, old_val)
 
```

Comment:
In `__init__`, we create properties with suffix `_old_`, so that we could know if they have changed when the user submitted a feature edit form from the frontend.

This won't work in the not-so-old ndb library as unlike the old DB Client Library, `__init__` creates instances with properties set to `None`. Hence, we could set up the old properties (`_old_prop_name`) in `_pre_put_hook()`. It seems to work properly.

Reply:
Hmm, do you see diffs being generated when notification emails are logged to the console after you edit a feature?  You will need to start the feature with one account and then edit it with a different account.

I might not be understanding when this hook is run, but I would think that the properties of the Feature object would have already been updated before put() is called, and thus before _pre_put_hook() is called.  That would cause the differences to always appear to be no difference.

I am thinking that _post_get_hook() would be the right hook.  But, that thought is just based on the names of the hook.  Only testing it will let us know for sure.

Reply:
I can see the following line showing ALL the properties of the  feature
> Starting to notify subscribers for feature {..._properties and values_..}

Is it the same diff that you were referring to ? 

I also tried to log to console the properties with and without the `_old_` suffix. It turns out that they have the same values. Hence, `_pre_put_hook()` will not work. Thank you for pointing this out. I did not notice it as there was one more error that was being caused by initializing properties in `__init()__` and the line `Starting to notify subscribers for feature` was same as the line generated by old DB Client Library. 

`_post_get_hook()` will not work as it is a class method unlike `_pre_put_hook()` which is an instance method. Hence, I have used the method `_from_pb()` which is responsible for assigning the values to the instance. See [this](https://github.com/GoogleChrome/chromium-dashboard/pull/1352/commits/3bfa050a8b582dfa79fa6c1a95e2263fa8aa8366) commit

Reply:
OK, that looks like it will work.